subsys/net/lib/dns/resolve.c

```diff
@@ -34,7 +34,7 @@ LOG_MODULE_REGISTER(net_dns_resolve, CONFIG_DNS_RESOLVER_LOG_LEVEL);
 #define LLMNR_IPV4_ADDR "224.0.0.252:5355"
 #define LLMNR_IPV6_ADDR "[ff02::1:3]:5355"
 
-#define DNS_BUF_TIMEOUT 500 /* ms */
+#define DNS_BUF_TIMEOUT K_MSEC(500) /* ms */
```

Comment:
I have to admit I didn't follow the timeout refactor (which would be 3rd or 4th in my count, or part of grande plan of which I already lost track). So in the very these commit, I see the changes like:

> #define DNS_TIMEOUT K_SECONDS(2)
> #define DNS_TIMEOUT (2 * MSEC_PER_SEC)

And then here:

> #define DNS_BUF_TIMEOUT 500 /* ms */
#define DNS_BUF_TIMEOUT K_MSEC(500) /* ms */

So, we replace `K_SECONDS()` with `* MSEC_PER_SEC` and yet introduce more uses of `K_MSEC()`?

I fail to understand nature and purpose of such changes. It all just goes in vicious circles, without clear purpose or improvement. My suggestions: a) if K_SECONDS() was removed, call to reinstate it; if it was deprecated, call to undeprecate it; b) skip doing purposeless changes like this, this will shrink this PR of following diffstats: +501 −376 to much more modest numbers. (Any line of changed lines may contain a typo, which we'll catch in a year due to lack of thorough testing).

Reply:
The timeouts in this example are used for different APIs. The buf timeout is needed for buffer allocation where the timeout type is k_timeout_t. The dns timeout is for timeout where the type is s32_t.